File: dhbw-automation/mobile/lib/core/services/travel_service.dart
import '../models/travel_models.dart';
import 'api_client.dart';

class TravelService {
  final ApiClient _apiClient;

  TravelService(this._apiClient);

  /// Holt Zugverbindungen basierend auf den Suchkriterien
  Future<TrainConnectionResponse> getConnections(
      TrainConnectionRequest request) async {
    try {
      final response = await _apiClient.post(
        '/api/travel/connections',
        data: request.toJson(),
      );
      return TrainConnectionResponse.fromJson(response.data);
    } catch (e) {
      throw Exception('Fehler beim Abrufen der Verbindungen: $e');
    }
  }

  /// Holt Standard-Route (Laupheim West → Ravensburg)
  Future<TrainConnectionResponse> getDefaultConnections({
    int maxConnections = 5,
  }) async {
    try {
      final response = await _apiClient.get(
        '/api/travel/connections/default',
        queryParameters: {'maxConnections': maxConnections},
      );
      return TrainConnectionResponse.fromJson(response.data);
    } catch (e) {
      throw Exception('Fehler beim Abrufen der Standard-Verbindungen: $e');
    }
  }
}

File: dhbw-automation/src/Frontend/src/services/travelService.ts
import api from './api';

export interface TrainConnectionRequest {
  from: string;
  to: string;
  dateTime?: string;
  maxConnections?: number;
}

export interface Journey {
  from: string;
  to: string;
  departure: string;
  arrival: string;
  duration: string;
  transfers: number;
  legs: Leg[];
  cancelled?: boolean;
  delay?: number;
}

export interface Leg {
  from: string;
  to: string;
  departure: string;
  arrival: string;
  line?: string;
  direction?: string;
  platform?: string;
  delay?: number;
  cancelled?: boolean;
}

export interface TrainConnectionResponse {
  journeys: Journey[];
  requestedAt: string;
}

export const travelService = {
  /**
   * Ruft Zugverbindungen ab
   */
  async getConnections(request: TrainConnectionRequest): Promise<TrainConnectionResponse> {
    const response = await api.post('/travel/connections', request);
    return response.data;
  },

  /**
   * Ruft die Standard-Verbindung (Laupheim West - Ravensburg) ab
   */
  async getDefaultConnections(maxConnections: number = 5): Promise<TrainConnectionResponse> {
    const response = await api.get(
      `/travel/connections/default?maxConnections=${maxConnections}`
    );
    return response.data;
  },
};
